crates/sui/src/upgrade_compatibility.rs

```diff
@@ -391,22 +505,47 @@ fn compare_packages(
         .map(|m| (m.self_id().name().to_owned(), m.clone()))
         .collect();
 
-    // for each existing find the new one run compatibility check
-    for existing_module in existing_modules {
-        let name = existing_module.self_id().name().to_owned();
-
-        // find the new module with the same name
-        match new_modules_map.get(&name) {
-            Some(new_module) => {
-                Compatibility::upgrade_check().check_with_mode::<CliCompatibilityMode>(
-                    &Module::new(&existing_module),
-                    &Module::new(new_module),
-                )?;
-            }
-            None => {
-                Err(anyhow!("Module {} is missing from the package", name))?;
+    let errors: Vec<String> = existing_modules
```

Comment:
the original version stopped on the first bad module, as per our discussion about not having users have to run the tool multiple times to get all of the errors I've reworked this to give all the errors from every module at once.